docs/mca/set-mca-param.rst

```diff
@@ -0,0 +1,245 @@
+run the application as:
+
+.. code-block:: sh
+
+   shell$ mpirun -np 2 a.out
+
+To use the ``foo.conf`` tuned parameter file, this command line
+changes to:
+
+.. code-block:: sh
+
+   shell$ mpirun -np 2 --tune foo.conf a.out
+
+Tuned parameter files can be coupled if more than one file is to be
+used. If there is another tuned parameter file called ``bar.conf``, it
+can be added to the command line as follows:
+
+.. code-block:: sh
+
+   shell$ mpirun -np 2 --tune foo.conf,bar.conf a.out
+
+The contents of tuned files consist of one or more lines, each of
+which contain zero or more `-x` and `--mca` options.  Comments are not
+allowed.  For example, the following tuned file:
+
+.. code-block::
+
+   -x envvar1=value1 -mca param1 value1 -x envvar2
+   -mca param2 value2
+   -x envvar3
+
+is equivalent to:
+
+.. code-block:: sh
+
+   shell$ mpirun \
+       -x envvar1=value1 -mca param1 value1 -x envvar2 \
+       -mca param2 value2
+       -x envvar3 \
+       ...rest of mpirun command line...
+
+Although the typical use case for tuned parameter files is to be
+specified on the command line, they can also be set as MCA parameters
+in the environment.  The MCA parameter ``mca_base_param_file_prefix``
+contains a comma-delimited list of tuned parameter files exactly as
+they would be passed to the ``--tune`` command line option.  The MCA
+parameter ``mca_base_param_file_path`` specifies the path to search
+for tuned files with relative paths.
+
+.. _label-mca-set-param-config-file:
+
+Configuration files
+-------------------
+
+Finally, simple configuration text files can be used to set MCA
+parameter values.  Parameters are set one per line (comments are
+permitted).  For example:
+
+.. code-block:: ini
+
+   # This is a comment
+   # Set the same MCA parameter as in previous examples
+   mpi_show_handle_leaks = 1
+
+Note that quotes are *not* necessary for setting multi-word values
+in MCA parameter files.  Indeed, if you use quotes in the MCA
+parameter file, they will be used as part of the value itself.  For
+example:
+
+.. code-block:: ini
+
+   # The following two values are different:
+   param1 = value with multiple words
+   param2 = "value with multiple words"
+
+By default, two files are searched (in order):
+
+#. ``$HOME/.openmpi/mca-params.conf``: The user-supplied set of
+   values takes the highest precedence.
+#. ``$prefix/etc/openmpi-mca-params.conf``: The system-supplied set
+   of values has a lower precedence.
+
+More specifically, the MCA parameter ``mca_param_files`` specifies a
+colon-delimited path of files to search for MCA parameters.  Files to
+the left have lower precedence; files to the right are higher
+precedence.
+
+.. note:: Keep in mind that, just like components, these parameter
+          files are *only* relevant where they are "visible"
+          (:ref:`see this FAQ entry
+          <faq-general-tuning-install-components>`).  Specifically,
+          Open MPI does not read all the values from these files
```

Comment:
Actually, this paragraph is untrue - we do read the param files and send the results everywhere. We specifically do _not_ read the files locally as that kills startup due to the filesystem pulling copies of the files everywhere. Pretty sure I stopped doing that quite some time ago, but definitely true for v5.

Reply:
@rhc54 do you mean that we read the param file once during startup and send them throughout the job? (Basically the opposite of what the paragraph says)

So the "per-node customization" is also completely wrong?

Reply:
Yes to both questions